minidump-common/src/format.rs

```diff
@@ -1389,33 +1389,74 @@ multi_structs! {
         pub build_string: [u16; 260], // MAX_PATH
         pub dbg_bld_str: [u16; 40],
     }
+
+    // Includes fields from MINIDUMP_MISC_INFO..4
+    /// Miscellaneous process and system information
+    ///
+    /// This struct matches the struct of the same name from minidumpapiset.h.
+    pub struct MINIDUMP_MISC_INFO5 {
+        pub xstate_data: XSTATE_CONFIG_FEATURE_MSC_INFO,
+        pub process_cookie: u32,
+    }
+    // TODO?: read xstate_data.enabled_features and process the extra XSAVE
+    // sections that should follow MISC_INFO_5 entries as documented in
+    // Volume 1, Chapter 13 of the "Intel 64 and IA-32 Architectures
+    // Software Developer’s Manual".
```

Comment:
This is a great idea. 

If I understand @luser's intentions correctly, he prefers to keep these types close to the binary format. Considering this, you could provide helper methods that look into `enabled_features` and then return an `Option<XStateFeature>` by `XStateFeatureIndex`.

Reply:
We'd like to implement that in a follow-up patch to keep complexity limited and also because we still have to gather a bunch of minidumps to test the feature (I've gathered minidumps  with 8 different combinations of XSTATE flags but they don't cover all of them).

Reply:
let's leave this for future work, since it potentially involves some restructuring of how we read this type out that I need to think through more.

Reply:
Could you outline your plans? It looks like all you need is one additional method that does roughly this:

```rust
impl XSTATE_CONFIG_FEATURE_MSC_INFO {
    pub fn feature(&self, index: XStateFeatureIndex) -> Option<XSTATE_FEATURE> {
        let index = index as usize;

        XStateFeatures::from_bits(1 << index)
            .filter(|f| self.enabled_features.contains(*f))
            .and_then(|_| self.features.get(index))
            .copied()
    }
}
```